it/v2/connectors/PersistenceConnectorSpec.scala

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.connectors
+
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import com.fasterxml.jackson.core.JsonParseException
+import com.github.tomakehurst.wiremock.client.WireMock.aResponse
+import com.github.tomakehurst.wiremock.client.WireMock.equalTo
+import com.github.tomakehurst.wiremock.client.WireMock.post
+import com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo
+import config.AppConfig
+import org.scalatest.concurrent.IntegrationPatience
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.http.ContentTypes
+import play.api.http.HeaderNames
+import play.api.http.Status.BAD_REQUEST
+import play.api.http.Status.INTERNAL_SERVER_ERROR
+import play.api.http.Status.OK
+import play.api.libs.json.JsSuccess
+import play.api.libs.json.Json
+import play.api.libs.ws.WSClient
+import play.mvc.Http.MimeTypes
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.UpstreamErrorResponse
+import utils.TestMetrics
+import v2.models.EORINumber
+import v2.models.MessageId
+import v2.models.MovementId
+import v2.models.errors.PresentationError
+import v2.models.errors.ErrorCode
+import v2.models.errors.StandardError
+import v2.models.responses.DeclarationResponse
+
+import java.nio.charset.StandardCharsets
+import scala.concurrent.ExecutionContext
+import scala.util.control.NonFatal
+
+class PersistenceConnectorSpec extends AnyFreeSpec with Matchers with GuiceOneAppPerSuite with utils.WiremockSuite with ScalaFutures with IntegrationPatience {
+
+  lazy val wsclient: WSClient   = app.injector.instanceOf[WSClient]
+  lazy val appConfig: AppConfig = app.injector.instanceOf[AppConfig]
+
+  lazy val persistenceConnector: PersistenceConnectorImpl = new PersistenceConnectorImpl(wsclient, appConfig, new TestMetrics())
+  implicit lazy val ec: ExecutionContext                  = app.materializer.executionContext
+
+  "POST /traders/:eori/message/departures" - {
+
+    lazy val okResult   = DeclarationResponse(MovementId("123"), MessageId("456"))
+    lazy val eoriNumber = EORINumber("ABC123")
```

Comment:
not required for this to be merged, but if you wanted to do a touch-up on this, you could take a look at Arbitrary - https://www.scala-exercises.org/scalacheck/arbitrary. Not that this needs to have multiple iterations, but randomly generated values are good